### PR TITLE
drivers/at86rf2xx: fix reliance on typical timings

### PR DESCRIPTION
**Pull request by immesys**
While doing testing of #5608 I discovered that the radio driver for the at86rf2xx uses typical timings for sleeps, and fails pretty badly if those timings are incorrect. In reality, state transition timings can be greater than the typical values quoted in the datasheet. For example:
- The temperature of the radio is different
- The capacitance on the radio oscillator is higher than expected due to parasitics, temperature or contamination
- The oscillator used to measure the state transition is running slow or has low precision.

In particular with #5608, the time intervals will be rounded down when converted to ticks. On some chips, the timer behind xtimer may be as slow as 16384 Hz (like on the Atmel SAM4L where the RTT cannot actually be used at 32khz) which means a tick is roughly 61us. 

This PR does a few things:
- Change one-off state transitions like reset to be longer than a tick. This is quite a bit longer, but resets are infrequent so I don't think the cost is significant.
- Round up WAKEUP_DELAY from 300 us to 306 us so that it goes from 4.9 ticks to 5 ticks for 16384 Hz systems
- Add logic to spin wait after WAKEUP_DELAY in assert_wake to compensate for the extra error that low power oscillators, temperature or contamination may introduce.

For the common case where 306 us was enough for the wakeup, this introduces no additional overhead.

In our testing deployment this fixed many weird radio errors that we got with #5608 pulled in.

### CODE REVIEW

File: drivers/at86rf2xx/at86rf2xx_netdev.c [hunk 1]
```diff
@@ -78,8 +78,8 @@ static int _init(netdev2_t *netdev)
     gpio_set(dev->params.reset_pin);
     gpio_init_int(dev->params.int_pin, GPIO_IN, GPIO_RISING, _irq_handler, dev);
 
-    /* make sure device is not sleeping, so we can query part number */
-    at86rf2xx_assert_awake(dev);
+    /* reset device to default values and put it into RX state */
+    at86rf2xx_reset(dev);
```
Comment (immesys) [inline drivers/at86rf2xx/at86rf2xx_netdev.c]: I forgot to mention this part. Previously, dev->state was set to SLEEP but after twiddling the gpio above it is actually in P_ON. If we go into assert_awake, that state mismatch means it waits for it to go into TRX_OFF but it won't because it is coming from P_ON not SLEEP. Assuming the chip state here was incorrect anyway, you should reset it before doing any communication.

File: drivers/include/at86rf2xx.h [hunk 2]
```diff
@@ -102,6 +102,7 @@ extern "C" {
  * @brief   Flags for device internal states (see datasheet)
  * @{
  */
+#define AT86RF2XX_STATE_P_ON         (0x00)     /**< initial power on */
```
Comment (immesys) [inline drivers/include/at86rf2xx.h]: Nope, actually it is needed

File: drivers/at86rf2xx/at86rf2xx_netdev.c (hunk 1 above)
Comment (OlegHahm) [inline drivers/at86rf2xx/at86rf2xx_netdev.c]: Shouldn't this go into a separate PR then?

**Comment by OlegHahm** [comment]
Nice job!

**Comment by immesys** [comment]
Rebased

File: drivers/at86rf2xx/at86rf2xx_netdev.c (hunk 1 above)
Comment (immesys) [inline drivers/at86rf2xx/at86rf2xx_netdev.c]: I can put it in a different PR, sure.

**Comment by OlegHahm** [comment]
I will test this next week - please remind me if I forget.

**Comment by PeterKietzmann** [comment]
@OlegHahm here comes your reminder to test this PR!

**Comment by immesys** [comment]
Having finally got time to get back to RIOT stuff, I have rebased this and fixed the style nit

**Comment by immesys** [comment]
Ah I didn't mention that I tested it and it was working fine in our deployments. I am about to rebase this again on the 2017.04 release for our own downstream release of low power RIOT.
